### ISSUE TITLE
Fail to deploy k8s cluster due to etcd health check

### ISSUE THREAD

**Issue by JosephKang** [microsoft/pai#4522]
**Organization Name**: Advantech

**Short summary about the issue/question**: Fail to deploy k8s cluster due to etcd health check

**Brief what process you are following**: 
- https://github.com/microsoft/pai/blob/v0.17.0/contrib/kubespray/doc/step-by-step-k8s.md#deploy-k8s
- ansible-playbook -i inventory/mycluster/host.yml cluster.yml --become --become-user=root -e "@inventory/mycluster/openpai.yml" --flush-cache

**How to reproduce it**:
- ansible-playbook -i inventory/mycluster/host.yml cluster.yml --become --become-user=root -e "@inventory/mycluster/openpai.yml" --flush-cache

**OpenPAI Environment**:
- OpenPAI version: 0.17.0
- Cloud provider or hardware configuration: Private OpenStack VMs
- OS (e.g. from /etc/os-release): ubuntu 16.04.6 
- Kernel (e.g. `uname -a`): 4.4.0-150-generic
- Hardware (e.g. core number, memory size, storage size, GPU type etc.):  Master: 8 Core, 64GB RAM
- Others: Kubespray v2.12.4

**Anything else we need to know**:
- ansible-playbook execution snapshot
`TASK [etcd : Configure | Check if etcd cluster is healthy] **********************************************************************************************************************************************************************************
Wednesday 13 May 2020  03:01:50 +0000 (0:00:00.047)       0:01:54.774 *********
FAILED - RETRYING: Configure | Check if etcd cluster is healthy (4 retries left).
FAILED - RETRYING: Configure | Check if etcd cluster is healthy (3 retries left).
FAILED - RETRYING: Configure | Check if etcd cluster is healthy (2 retries left).
FAILED - RETRYING: Configure | Check if etcd cluster is healthy (1 retries left).
fatal: [master-970696-iaas -> 192.168.211.6]: FAILED! => {"attempts": 4, "changed": false, "cmd": "/usr/local/bin/etcdctl --no-sync --endpoints=https://192.168.211.6:2379 cluster-health | grep -q 'cluster is healthy'", "delta": "0:00:00.019542", "end": "2020-05-13 03:02:15.407216", "msg": "non-zero return code", "rc": 1, "start": "2020-05-13 03:02:15.387674", "stderr": "Error:  client: etcd cluster is unavailable or misconfigured; error #0: dial tcp 192.168.211.6:2379: connect: connection refused\n\nerror #0: dial tcp 192.168.211.6:2379: connect: connection refused", "stderr_lines": ["Error:  client: etcd cluster is unavailable or misconfigured; error #0: dial tcp 192.168.211.6:2379: connect: connection refused", "", "error #0: dial tcp 192.168.211.6:2379: connect: connection refused"], "stdout": "", "stdout_lines": []}
PLAY RECAP **********************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0
master-970696-iaas         : ok=428  changed=30   unreachable=0    failed=1
worker-970684-iaas         : ok=285  changed=15   unreachable=0    failed=0
`
- Already use **ntp** to sync master and worker clock
- Manual etcd health check result
`etcdctl --debug cluster-health
Cluster-Endpoints: http://127.0.0.1:2379, http://127.0.0.1:4001
cURL Command: curl -X GET http://127.0.0.1:2379/v2/members
cURL Command: curl -X GET http://127.0.0.1:4001/v2/members
cluster may be unhealthy: failed to list members
Error:  client: etcd cluster is unavailable or misconfigured
; error #0: dial tcp 127.0.0.1:2379: connect: connection refused
; error #1: dial tcp 127.0.0.1:4001: connect: connection refused
error #0: dial tcp 127.0.0.1:2379: connect: connection refused
error #1: dial tcp 127.0.0.1:4001: connect: connection refused`

**Comment by JosephKang** [comment]
It seems that etcd docker container creation failed.
- `ubuntu@master-970696-iaas:~/kubespray⟫ date; docker ps -a
Wed May 13 03:48:34 UTC 2020
CONTAINER ID        IMAGE                         COMMAND                 CREATED             STATUS              PORTS               NAMES
8fa42d2f4efb        quay.io/coreos/etcd:v3.3.12   "/usr/local/bin/etcd"   1 second ago        Created                                 etcd1
- ubuntu@master-970696-iaas:~/kubespray⟫ date; docker ps -a
Wed May 13 03:49:44 UTC 2020
CONTAINER ID        IMAGE                         COMMAND                 CREATED             STATUS              PORTS               NAMES
045b36bdc144        quay.io/coreos/etcd:v3.3.12   "/usr/local/bin/etcd"   9 seconds ago       Created                                 etcd1
`

**Comment by ydye** [comment]
You only have 1 master node , right.

**Comment by ydye** [comment]
Try to this command:
```
sudo journalctl -u etcd | tail200
```

**Comment by ydye** [comment]
And we have came accorss the similar issue before. We solve it by the following task.

- rest deployment
- clean all container on the node
- restart docker daemon
- and reboot the vm

**Comment by JosephKang** [comment]
> You only have 1 master node , right?
Yes

> sudo journalctl -u etcd | tail -n 200
`
...
May 13 06:07:52 master-970696-iaas systemd[1]: Started etcd docker wrapper.
May 13 06:07:53 master-970696-iaas etcd[19714]: /usr/bin/docker: Error response from daemon: OCI runtime create failed: unable to retrieve OCI runtime error (open /run/containerd/io.containerd.runtime.v1.linux/moby/9090ceee1feb868825e4df09d9202276a8d79b7d3238e1da959d116717e50ee2/log.json: no such file or directory): fork/exec /usr/bin/nvidia-container-runtime: no such file or directory: unknown.
May 13 06:07:53 master-970696-iaas etcd[19714]: time="2020-05-13T06:07:53Z" level=error msg="error waiting for container: context canceled"
May 13 06:07:53 master-970696-iaas systemd[1]: etcd.service: Main process exited, code=exited, status=127/n/a
May 13 06:07:53 master-970696-iaas docker[19741]: etcd1
May 13 06:07:53 master-970696-iaas systemd[1]: etcd.service: Unit entered failed state.
May 13 06:07:53 master-970696-iaas systemd[1]: etcd.service: Failed with result 'exit-code'.
May 13 06:08:08 master-970696-iaas systemd[1]: etcd.service: Service hold-off time over, scheduling restart.
May 13 06:08:08 master-970696-iaas systemd[1]: Stopped etcd docker wrapper.
May 13 06:08:08 master-970696-iaas systemd[1]: Starting etcd docker wrapper...
May 13 06:08:08 master-970696-iaas docker[19973]: etcd1
May 13 06:08:08 master-970696-iaas systemd[1]: Started etcd docker wrapper.
May 13 06:08:08 master-970696-iaas etcd[19986]: /usr/bin/docker: Error response from daemon: OCI runtime create failed: unable to retrieve OCI runtime error (open /run/containerd/io.containerd.runtime.v1.linux/moby/a0640e56fec7c52c17739bba1199f9da0b3759a4a7a527813738a5d5fc217b6d/log.json: no such file or directory): fork/exec /usr/bin/nvidia-container-runtime: no such file or directory: unknown.
May 13 06:08:08 master-970696-iaas etcd[19986]: time="2020-05-13T06:08:08Z" level=error msg="error waiting for container: context canceled"
May 13 06:08:08 master-970696-iaas systemd[1]: etcd.service: Main process exited, code=exited, status=127/n/a
May 13 06:08:08 master-970696-iaas docker[20011]: etcd1
May 13 06:08:08 master-970696-iaas systemd[1]: etcd.service: Unit entered failed state.
May 13 06:08:08 master-970696-iaas systemd[1]: etcd.service: Failed with result 'exit-code'.
`

**Comment by ydye** [comment]
It's a docker issue. Which branch do you use.

**Comment by JosephKang** [comment]
> It's a docker issue. Which branch do you use.

docker version? 19.03.8
Kubespray : v2.12.4
openpai: v0.17.0

**Comment by ydye** [comment]
we haven't test kubespray 2.11 before, so from kubespary part we can't gurantee the compatibility. Suggest to switch to 2.11.
In openpai v0.17
https://github.com/microsoft/pai/blob/pai-0.17.y/contrib/kubespray/quick-start/openpai.yml#L343
https://github.com/microsoft/pai/blob/pai-0.17.y/contrib/kubespray/daemon-openpai-nvidia-runtime.json
We assue worker node installed with nvidia-runtimue. You could install it on your worker node.

I believe the etcd issue could be fixed after installing nvidia-runtime.
https://github.com/microsoft/pai/blob/pai-0.17.y/contrib/kubespray/nvidia-docker.yml

**Comment by JosephKang** [comment]
@ydye,

How to install nvidia-runtime on worker manually?
- `
ubuntu@worker-970684-iaas:> nvidia-smi
Wed May 13 15:07:29 2020
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 418.56       Driver Version: 418.56       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000000:00:05.0 Off |                    0 |
| N/A   32C    P0    36W / 300W |      0MiB / 32480MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

`

**Comment by ydye** [comment]
https://github.com/NVIDIA/nvidia-container-runtime

**Comment by JosephKang** [comment]
- Install `nvidia-container-runtime` on worker machine
- checkout kubespray release-2.11
- ansible-playbook reset k8s deployment
- restart docker
- reboot master and worker VMs
- ansible-playbook deply openpai.yml
- Still the same etcd error: sudo journalctl -u etcd | tail -n 200
May 13 10:41:29 master-970696-iaas systemd[1]: etcd.service: Failed with result 'exit-code'.
May 13 10:41:44 master-970696-iaas systemd[1]: etcd.service: Service hold-off time over, scheduling restart.
May 13 10:41:44 master-970696-iaas systemd[1]: Stopped etcd docker wrapper.
May 13 10:41:44 master-970696-iaas systemd[1]: Starting etcd docker wrapper...
May 13 10:41:44 master-970696-iaas docker[19137]: etcd1
May 13 10:41:44 master-970696-iaas systemd[1]: Started etcd docker wrapper.
May 13 10:41:44 master-970696-iaas etcd[19149]: /usr/bin/docker: Error response from daemon: OCI runtime create failed: unable to retrieve OCI runtime error (open /run/containerd/io.containerd.runtime.v1.linux/moby/a5d5a0bef26349a67fca916ed8af33095ce5e9dd822699475ee9aa34ea4c30c8/log.json: no such file or directory): fork/exec /usr/bin/nvidia-container-runtime: no such file or directory: unknown.
May 13 10:41:44 master-970696-iaas etcd[19149]: time="2020-05-13T10:41:44Z" level=error msg="error waiting for container: context canceled"
May 13 10:41:44 master-970696-iaas systemd[1]: etcd.service: Main process exited, code=exited, status=127/n/a
May 13 10:41:45 master-970696-iaas docker[19175]: etcd1
May 13 10:41:45 master-970696-iaas systemd[1]: etcd.service: Unit entered failed state.
May 13 10:41:45 master-970696-iaas systemd[1]: etcd.service: Failed with result 'exit-code'.

**Comment by abuccts** [comment]
could you paste following logs of `dockerd` on your worker node:
```sh
sudo systemctl status docker.service
sudo docker info
```
and check `nvidia-container-runtime` is installed correctly:
```sh
whereis nvidia-container-runtime
```
you probably didn't install `nvidia-container-runtime` or uninstalled it by mistake

**Comment by JosephKang** [comment]
**sudo systemctl status docker.service**
- docker.service - Docker Application Container Engine
   Loaded: loaded (/etc/systemd/system/docker.service; enabled; vendor preset: enabled)
  Drop-In: /etc/systemd/system/docker.service.d
           └─docker-dns.conf, docker-options.conf
   Active: active (running) since Wed 2020-05-13 18:36:42 CST; 4h 20min ago
     Docs: http://docs.docker.com
 Main PID: 24817 (dockerd)
    Tasks: 14
   Memory: 30.0M
      CPU: 7.011s
   CGroup: /system.slice/docker.service
           └─24817 /usr/bin/dockerd --config-file /etc/docker-openpai/daemon-openpai.json --data-root=/mnt/docker --log-opt max-size=2g --log-opt max-file=2 --log-drive
May 13 18:36:42 worker-970684-iaas dockerd[24817]: time="2020-05-13T18:36:42.093513350+08:00" level=warning msg="Your kernel does not support swap memory limit"
May 13 18:36:42 worker-970684-iaas dockerd[24817]: time="2020-05-13T18:36:42.093546409+08:00" level=warning msg="Your kernel does not support cgroup rt period"
May 13 18:36:42 worker-970684-iaas dockerd[24817]: time="2020-05-13T18:36:42.093553222+08:00" level=warning msg="Your kernel does not support cgroup rt runtime"
May 13 18:36:42 worker-970684-iaas dockerd[24817]: time="2020-05-13T18:36:42.093844516+08:00" level=info msg="Loading containers: start."
May 13 18:36:42 worker-970684-iaas dockerd[24817]: time="2020-05-13T18:36:42.104539019+08:00" level=info msg="Default bridge (docker0) is assigned with an IP address 17
May 13 18:36:42 worker-970684-iaas dockerd[24817]: time="2020-05-13T18:36:42.113290594+08:00" level=info msg="Loading containers: done."
May 13 18:36:42 worker-970684-iaas dockerd[24817]: time="2020-05-13T18:36:42.139457738+08:00" level=info msg="Docker daemon" commit=2d0083d graphdriver(s)=overlay2 vers
May 13 18:36:42 worker-970684-iaas dockerd[24817]: time="2020-05-13T18:36:42.139503552+08:00" level=info msg="Daemon has completed initialization"
May 13 18:36:42 worker-970684-iaas dockerd[24817]: time="2020-05-13T18:36:42.144275228+08:00" level=info msg="API listen on /var/run/docker.sock"
May 13 18:36:42 worker-970684-iaas systemd[1]: Started Docker Application Container Engine.

**sudo docker info**
- Client:
 Debug Mode: false
Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 9
 Server Version: 18.09.7
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 4.4.0-178-generic
 Operating System: Ubuntu 16.04.6 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 88.51GiB
 Name: worker-970684-iaas
 ID: GAHX:GNXM:A7GZ:UFVH:DCXL:F73U:IIXE:FQBN:4XAH:62CF:CPPD:TA5Z
 Docker Root Dir: /mnt/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine

**whereis nvidia-container-runtime**
- nvidia-container-runtime: /usr/bin/nvidia-container-runtime /etc/nvidia-container-runtime

**Comment by abuccts** [comment]
Here're the problems:
1. on `master-970696-iaas` node, `nvidia-container-runtime` is not installed but `dockerd` includes it.
2. on `worker-970684-iaas` node, `nvidia-container-runtime` is installed but `dockerd` does not include it.

The first one caused etcd container cannot be started, the second one will cause GPUs cannot be used in jobs.

---

Here're the solutions:
1. if there's no GPU on master node, remove nvidia runtime in docker daemon config, which may locate in `/etc/docker-openpai/daemon-openpai.json` or `/etc/docker/daemon.json`.
2. on worker node with GPU, add nvidia runtime to docker daemon config.

the nvidia runtime config has two fields in json, you should restart docker service after changing config.
```
{
  ...
  "default-runtime": "nvidia",
  "runtimes": {
      "nvidia": {
          "path": "/usr/bin/nvidia-container-runtime",
          "runtimeArgs": []
      }
  }
}
```

**Comment by JosephKang** [comment]
1. The `/etc/docker-openpai/daemon-openpai.json` only contain storage-driver configuration and /etc/docker/daemon.json is empty. Should I remove or disable nvidia runtime in kubespray cluster.yml or openpai.yml template?

2. Thanks. I will add the configuration to /etc/docker-openpai/daemon-openpai.json and redeploy the cluster.

**Comment by JosephKang** [comment]
Execute another kubespray k8s deployment. 
On master, I can see etcd instance under docker but cannot see etcd kube pod
- ubuntu@pai2master-971908-iaas:~/kubespray⟫ docker ps | grep etcd
65016af343f6        quay.io/coreos/etcd:v3.3.10                           "/usr/local/bin/etcd"    29 minutes ago      Up 28 minutes                           etcd1
-  ubuntu@pai2master-971908-iaas:/kubespray⟫ kubectl get pods --all-namespaces | grep etc
1 ubuntu@pai2master-971908-iaas:/kubespray⟫

The etcd docker log information shows that some connection locally is rejected so I am not sure if the etcd is started correctly on master.
- 2020-05-14 01:01:06.731894 I | embed: rejected connection from "192.168.211.11:49172" (error "EOF", ServerName "")
2020-05-14 01:06:33.809841 I | mvcc: store.index: compact 883
2020-05-14 01:06:33.813388 I | mvcc: finished scheduled compaction at 883 (took 3.20463ms)
2020-05-14 01:11:33.812126 I | mvcc: store.index: compact 1376
2020-05-14 01:11:33.813027 I | mvcc: finished scheduled compaction at 1376 (took 657.901µs)

**Comment by ydye** [comment]
- please share your openpai.yml
- please share the content of this file: ```/etc/systemd/system/docker.service.d/docker-options.conf```

**Comment by JosephKang** [comment]
- [openpai.yml.txt](https://github.com/microsoft/pai/files/4625976/openpai.yml.txt)
- [docker-options.conf.txt](https://github.com/microsoft/pai/files/4625977/docker-options.conf.txt)

**Comment by ydye** [comment]
I haven't find any issue in the configuration
Can you paste the dockerd’s log with the command?
```
sudo journalctl -u docker | tail -n200
```

**Comment by JosephKang** [comment]
-

**Comment by JosephKang** [comment]
Restart master docker since 06:55:50 UTC: [master_restart_docker.txt](https://github.com/microsoft/pai/files/4626547/master_restart_docker.txt)

**Comment by JosephKang** [comment]
I found one of the issue in my env: The worker is set to CST instead of UTC.
The redeployment seems OK after changing the timezone in worker host to UTC and resetting the cluster.
- deploy result
PLAY RECAP 
localhost                  : ok=1    changed=0    unreachable=0    failed=0
pai2master-971908-iaas     : ok=642  changed=88   unreachable=0    failed=0
pai2worker-971911-iaas     : ok=390  changed=49   unreachable=0    failed=0
- etcd docker logs
2020-05-14 07:14:13.246636 I | etcdmain: etcd Version: 3.3.10
2020-05-14 07:14:13.246638 I | etcdmain: Git SHA: 27fc7e2
2020-05-14 07:14:13.246641 I | etcdmain: Go Version: go1.10.4
2020-05-14 07:14:13.246643 I | etcdmain: Go OS/Arch: linux/amd64
2020-05-14 07:14:13.246646 I | etcdmain: setting maximum number of CPUs to 8, total number of available CPUs is 8
2020-05-14 07:14:13.246682 W | etcdmain: found invalid file/dir .bash_logout under data dir /var/lib/etcd (Ignore this if you are upgrading etcd)
2020-05-14 07:14:13.246685 W | etcdmain: found invalid file/dir .bashrc under data dir /var/lib/etcd (Ignore this if you are upgrading etcd)
2020-05-14 07:14:13.246687 W | etcdmain: found invalid file/dir .profile under data dir /var/lib/etcd (Ignore this if you are upgrading etcd)
2020-05-14 07:14:13.246693 N | etcdmain: the server is already initialized as member before, starting as etcd member...
2020-05-14 07:14:13.246709 I | embed: peerTLS: cert = /etc/ssl/etcd/ssl/member-pai2master-971908-iaas.pem, key = /etc/ssl/etcd/ssl/member-pai2master-971908-iaas-key.pem, ca = , trusted-ca = /etc/ssl/etcd/ssl/ca.pem, client-cert-auth = true, crl-file =
2020-05-14 07:14:13.247144 I | embed: listening for peers on https://192.168.211.11:2380
2020-05-14 07:14:13.247177 I | embed: listening for client requests on 127.0.0.1:2379
2020-05-14 07:14:13.247195 I | embed: listening for client requests on 192.168.211.11:2379
2020-05-14 07:14:13.247788 I | etcdserver: name = etcd1
2020-05-14 07:14:13.247801 I | etcdserver: data dir = /var/lib/etcd
2020-05-14 07:14:13.247805 I | etcdserver: member dir = /var/lib/etcd/member
2020-05-14 07:14:13.247807 I | etcdserver: heartbeat = 250ms
2020-05-14 07:14:13.247810 I | etcdserver: election = 5000ms
2020-05-14 07:14:13.247812 I | etcdserver: snapshot count = 10000
2020-05-14 07:14:13.247820 I | etcdserver: advertise client URLs = https://192.168.211.11:2379
2020-05-14 07:14:13.248328 I | etcdserver: restarting member 1ceb938d1dac85e8 in cluster 182863484b8e4cea at commit index 5
2020-05-14 07:14:13.248717 I | raft: 1ceb938d1dac85e8 became follower at term 2
2020-05-14 07:14:13.248733 I | raft: newRaft 1ceb938d1dac85e8 [peers: [], term: 2, commit: 5, applied: 0, lastindex: 5, lastterm: 2]
2020-05-14 07:14:13.249765 W | auth: simple token is not cryptographically signed
2020-05-14 07:14:13.256236 I | etcdserver: starting server... [version: 3.3.10, cluster version: to_be_decided]
2020-05-14 07:14:13.256578 I | etcdserver/membership: added member 1ceb938d1dac85e8 [https://192.168.211.11:2380] to cluster 182863484b8e4cea
2020-05-14 07:14:13.256650 N | etcdserver/membership: set the initial cluster version to 3.3
2020-05-14 07:14:13.256681 I | etcdserver/api: enabled capabilities for version 3.3
2020-05-14 07:14:13.258061 I | embed: ClientTLS: cert = /etc/ssl/etcd/ssl/member-pai2master-971908-iaas.pem, key = /etc/ssl/etcd/ssl/member-pai2master-971908-iaas-key.pem, ca = , trusted-ca = /etc/ssl/etcd/ssl/ca.pem, client-cert-auth = true, crl-file =
2020-05-14 07:14:20.749231 I | raft: 1ceb938d1dac85e8 is starting a new election at term 2
2020-05-14 07:14:20.749282 I | raft: 1ceb938d1dac85e8 became candidate at term 3
2020-05-14 07:14:20.749326 I | raft: 1ceb938d1dac85e8 received MsgVoteResp from 1ceb938d1dac85e8 at term 3
2020-05-14 07:14:20.749350 I | raft: 1ceb938d1dac85e8 became leader at term 3
2020-05-14 07:14:20.749364 I | raft: raft.node: 1ceb938d1dac85e8 elected leader 1ceb938d1dac85e8 at term 3
2020-05-14 07:14:20.749805 I | etcdserver: published {Name:etcd1 ClientURLs:[https://192.168.211.11:2379]} to cluster 182863484b8e4cea
2020-05-14 07:14:20.749960 I | embed: ready to serve client requests
2020-05-14 07:14:20.750119 I | embed: ready to serve client requests
2020-05-14 07:14:20.754095 I | embed: serving client requests on 127.0.0.1:2379
2020-05-14 07:14:20.754165 I | embed: serving client requests on 192.168.211.11:2379
proto: no coders for int
proto: no encoder for ValueSize int [GetProperties]
